### ISSUE TITLE
Team update - Oct 05, 2022

### ISSUE THREAD

**Issue by trallard** [Quansight-Labs/jupyter-a11y-mgmt#161]
This is a @Quansight-Labs/czi-a11y-grant team sync 🎉🎉🎉! This is a way for the Team Members to provide status reports on what they've been up to this week and request help and attention for things they are working on. This issue will be closed at the end of the day.

_Copy and paste the template below, and answer questions as you wish!_

<details>
<summary>Response Template</summary>

```md
**Thanks I'd like to give 🙌**
- So-and-so helped me out a lot with foo...

**Updates from last week :heavy_check_mark:**
- I worked towards deliverable: <link-to-deliverable>

**Challenges I faced and things I'd like assistance with 🙏**
- I had a hard time figuring out ...

**My availability for next week**
- I'll be off on foo day...

**Important items for discussion 💬**
- I have a question about goal <link-to-goal>
- Can @foo give a comment on issue #NN ?
- I opened #NN for discussion, please chime in
- I need to discuss foo
```

</details>

#### :mag: Needs Triage

_The [Needs Triage issues](https://github.com/Quansight-Labs/jupyter-a11y-mgmt/labels/status%3A%20needs%20triage) require an initial assessment and labeling._

No issues need triage! 🎉

### :dart: Project boards

Please make sure the boards reflect the current status of the project.

[Jupyter a11y grant project](https://github.com/orgs/Quansight-Labs/projects/5/views/5)

**Comment by steff456** [comment]
**Updates from last week :heavy_check_mark:**
- I worked in the accessibility pygments package
    - Created a demo html to see the themes in multiple languages at a time
    - Made a major refactoring in order to add themes easily
    - Created a contribution guide

**Challenges I faced and things I'd like assistance with 🙏**
- I'm ok

**My availability for next week**
- I'll be here

**Important items for discussion 💬**
- I don't have anything important for today

**Comment by isabela-pf** [comment]
**Thanks I'd like to give 🙌**
- @steff456's theming work is inspiring!

**Updates from last week :heavy_check_mark:**
- I don't think I made substantial progress on any of the main things on my plate yet.
- I set some next steps for myself with 200% zoom in slack.
- I reviewed and merged my keyboard navigation workshop PR since there haven't been any updates. Hoping to give it a test soon.

**Challenges I faced and things I'd like assistance with 🙏**

**My availability for next week**
- I'll be here all week.

**Important items for discussion 💬**
- I don't think there's anything we can discuss.

**Comment by trallard** [comment]
@kulsoomzahra 
	- demo of solution https://github.com/Quansight-Labs/jupyterlab-accessible-themes/issues/37

--- 
@isabela-pf 
    - Merged the keyboard navigation workshop
    - Zoom: Command palette and Tab to be added 
    - Actions: will work on typing the proposals for access

--- 
@steff456 
	- completed the first iteration of the `a11y-pygments` package
	- will work on the release on pypi and conda

---
@gabalafou 
	- rethinking testing and reporting 
	- as we are working on fixing issues -> e.g. colour contrast -> work on figure out to add that as a test (manual + automated where possible) 
	- can we create "personas"? https://alphagov.github.io/accessibility-personas/setup/chromebook.html#setting-up-personas-on-chromebooks

**Comment by gabalafou** [comment]
Sorry I'm late on this. I moved some meetings around without leaving myself time to write my update before the meeting.

**Thanks I'd like to give 🙌**
- Thanks to @kulsoomzahra for sharing about accessibility, her work and ours in the internship QShare on Monday.
- Thanks to @isabela-pf for sharing the [Eric Bailey article about accessibility being a backend concern](https://ericwbailey.website/published/yes-accessibility-is-also-a-backend-concern/). It led me to [another article of his](https://www.smashingmagazine.com/2021/09/improving-accessibility-of-markdown/) that had a quote that I found helpful:

> The thing about accessibility work is that every little bit helps. You don’t have to address every consideration I have in this post in one big, sweeping change. Instead, try picking one thing to focus on, and build from there.
 
**Updates from last week :heavy_check_mark:**
- I spent about half my time working on WCAG 1.4.10, thinking about ways to possibly meet it, providing feedback on [Isabela's proposals](https://github.com/Quansight-Labs/jupyterlab-accessible-themes/issues/34#issuecomment-1262458172).
- The other half of my time I spent thinking hard about the testing repo and what purpose it serves. I've been wanting to work towards something that can be considered a full end-to-end product, and I have an idea, which I will add in a later comment to this issue.

**Challenges I faced and things I'd like assistance with 🙏**
- I've once again lost sight of what the purpose of the testing repo is. I need to be given the elevator pitch again. 
 
**My availability for next week**
- I'll be here.

**Comment by gabalafou** [comment]
@kulsoomzahra I’m not sure if you saw this in the team meeting, but I discovered that the design we were discussing for the running tabs panel has already been implemented in Notebook 7. To see it, follow the Binder link at the Discourse thread [announcing the Notebook 7 pre-release](https://discourse.jupyter.org/t/notebook-7-pre-releases-are-available/16063). You can then take a look at the Running tab. :)

**Comment by gabalafou** [comment]
@trallard, here's my follow up to our conversation, my proposal for the rest of this month. Here's what I would like my deliverable for the end of this month to be. I want to create a working service such that when I open a PR that fixes an accessibility issue on JupyterLab, I can ask the service to run a set of tests and post a comment on the PR that show 1 failing test before the PR and all tests passing after the PR. 

For a first example, I will use my [tab trap test](https://github.com/Quansight-Labs/jupyter-a11y-testing/blob/main/testing/jupyterlab/tests/tab-loop-no-trap.test.ts) plus [fix](https://github.com/jupyterlab/lumino/pull/373). I will write up a document (in markdown) explaining this test and how to perform it manually. I will set up the testing service so that it can run only tests marked as regression tests. I will add a feature to the testing service that will allow it to perform a before/after comparison, highlighting failed tests and providing links to the markdown documentation for each failed test. 

In the end, my demo of this deliverable will be a PR on my fork of JupyterLab. On that PR, you will see a comment from the testing service that shows one test (tab trap) failing before the PR, and passing after the PR. In that comment, you will get a link to the markdown document that explains the failing tab trap test.

To recap, here are the components of this deliverable:

- Add markdown doc for the one tab trap test. This will demonstrate a system that builds upon Isabela's test script template to create individual markdown files for each test. The [ACT Rules collection](https://act-rules.github.io/rules/) is somewhat similar to what I want to build towards, but instead of rules, we'll be documenting actual tests that can be performed manually and in some cases automatically.
- Add ability to build JupyterLab from source and run tests against it.
- Add ability to run a subset of tests marked as regressions.
- Add ability to take a JupyterLab PR (whether on a fork or on the main repo) as input, and produce as output a comment on that PR that compares a regression test run on that PR's parent branch versus the PR's branch, providing for each test failure a link to that test's documentation in our GitHub testing repo.

**Comment by gabalafou** [comment]
@trallard to respond specifically to your question about using personas, I have to think some more about it. The blog post frames the personas as being useful in the design process, doesn't say anything about using them in the testing process, but maybe there is a way to somehow incorporate the general idea into a testing framework. I'll have to think about that some more.
